contributors/design-proposals/scheduling/pod-preemption.md

```diff
@@ -106,59 +107,295 @@ Now, assume everything in the above example, but the best effort pod has priorit
+the failure is due to pod affinity.
+*   If there are many pods that match one or more affinity rules of P (step 4)
+their permutation may produce a large set. Trying them all in step 6 may cause
+performance degradation.
+
+
+### Decision
+
+Supporting inter-pod affinity on lower priority pods needs a fairly complex logic
+which could degrade performance when there are many pods matching the pending
+pod's affinity rules. We could have limited the maximum number of matching pods
+supported in order to address the performance issue, but it would have been very
+confusing to users and would have removed predictability of scheduling. Moreover,
+inter-pod affinity is a way for users to define dependency among pods. Inter-pod
+affinity to lower priority pods creates dependency on lower priority pods. Such
+a dependency is probably not desired in most realistic scenarios. Given these
+points, we decided not to implement this feature.
+
+## Supporting Cross Node Preemption?
+
+In certain scenarios, scheduling a pending pod (P) on a node (N1), requires
+preemption of one or more pods on other nodes. An example of such scenarios is a
+lower priority pod with anti-affinity to P running on a different node in the same zone and the 
+topology of the anti-affinity is zone. Another example is a lower priority pod
+running on a different node than N1 and is consuming a non-local resource that P
+needs. In all of such cases, preemption of one or more pods on nodes other than
+N1 is required to make P schedulable on N1. Such a preemption is called "cross
+node preemption".
+
+### Potential Solution
+
+When a pod P is not schedulable on a node N even after removal of all lower
+priority pods from node N, there may be other pods on other nodes that are not
+allowing it to schedule. Since scheduler preemption logic should not rely on 
+the internals of its predicate functions, it has to perform an exhaustive search
+for other pods whose removal may allow P to be scheduled. Such an exhaustive
+search will be prohibitively expensive in large clusters.
+
+### Decision
+
+Given that we do not have a solution with reasonable performance for supporting
+cross node preemption, we have decided not to implement this feature.
+
+# Interactions with Cluster Autoscaler
+
+Preemption gives higher precedence to most important pods in the cluster and
+tries to provide better availability of cluster resources for such pods. As a
+result, we may not need to scale the cluster up for all pending pods. Particularly,
+scaling up the cluster may not be necessary in two scenarios:
+
+1. The pending pod has already preempted pods and is going to run on a node soon.
+1. The pending pod is very low priority and the owner of the cluster prefers to save
+money by not scaling up the cluster for such a pod.
+
+In order to address these cases:
+1. Cluster Autoscaler will not scale up the cluster for pods with
+`NominatedNodeName` annotation.
```

Comment:
Technically speaking I think we need to go a step further and assume that the pod with `NominatedNodeName` is already scheduled on the nominated node, thus preventing scale-down of this node as well. Though this is more of an implementation detail in CA and doesn't really matter much in context of this proposal.

Reply:
I was under the impression that autoscaler does not scale down a cluster that has unschedulable pending pods. Does it actually scale down in presence of such pending pods? If so, then what you said is valid and it shouldn't scale down.